apps/web/src/components/PoolCandidatesTable/PoolCandidatesTable.tsx

```diff
@@ -378,33 +379,69 @@ const PoolCandidatesTable = ({
     ?.filter(notEmpty)
     .map((skill) => skill.id);
 
-  const querySelected = async (action: SelectingFor) => {
-    setSelectingFor(action);
-    setIsSelecting(true);
-    return client
-      .query(PoolCandidatesTable_SelectPoolCandidatesQuery, {
-        ids: selectedRows,
-      })
-      .toPromise()
-      .then((result) => {
-        const poolCandidates: PoolCandidate[] = unpackMaybes(
-          result.data?.poolCandidates,
+  const isPoolCandidate = (
+    candidate: Error | PoolCandidate | null,
+  ): candidate is PoolCandidate =>
+    candidate !== null && !(candidate instanceof Error);
+
+  const batchLoader = new DataLoader<string, PoolCandidate | null>(
+    async (ids) => {
+      const batchSize = 8;
```

Comment:
Is there a reason we settled on 8? This results in many requests that tend to block each other and create a very long waterfall. Is there any danger to increase this number so we can make fewer requests? Especially since we are on HTTP1 which has some trouble with multiple requests blocking.

![Screenshot 2024-02-20 104627](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/1332c57c-fe57-4aa3-847d-21f044a74cf8)

Reply:
yeah I explained in PR details. Dev was fine with 50 for safer and even good with 100 records per batch with little memory spike.

Reply:
That's strange, I set batch to 100. 
Pulling 221 candidates

Query 1: 2.13 MB in 20 seconds

Query 2: 2.14 MB in 19.7 seconds

Query 3: 0.5 MB in 7.3 seconds

Excel document of 221 candidates size 280 KB successful

Ten shouldn't be causing issues 😕

Reply:
I beleive it is server dependant so when run locally it all depends on how much your computer can handle. I'm wondering if this calls for a env var so we can set it higher in prod but allow devs to set it to a reasonable number for local testing? 🤔 

Or is that overkill and we just set it at a prod number and put a comment to tell devs to lower manually if needed?

Reply:
Not sure, it can be handy for deployed too for fidgeting with the number much more quickly. Adjusting both up and down.

Reply:
Exactly what I was thinking just now Eric!
Env variable would be easier for long run.

But if we think we are not keeping this batching for long run, not worth adding an env variable.

I will test again with 100 on dev. to make sure.

But I'm super impressed on Vachan's machine performance. 🫠